src/algebra/geom_sum.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2019 Neil Strickland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Neil Strickland
+
+Sums of finite geometric series
+-/
+
+import algebra.big_operators algebra.commute
+
+universe u
+variable {α : Type u}
+
+open finset
+
+def geom_series [semiring α] (x : α) (n : ℕ) :=
+(range n).sum (λ i, x ^ i)
+
+theorem geom_series_def [semiring α] (x : α) (n : ℕ) :
+  geom_series x n = (range n).sum (λ i, x ^ i) := rfl
+
+def geom_series₂ [semiring α] (x y : α) (n : ℕ) :=
+(range n).sum (λ i, x ^ i * (y ^ (n - 1 - i)))
+
+theorem geom_series₂_with_one [semiring α] (x : α) (n : ℕ) :
```

Comment:
This should probably be tagged with `@[simp]`.

Reply:
Done.